design-documents/graph-ql/coverage/dynamic-blocks.graphqls

```diff
@@ -0,0 +1,42 @@
+type DynamicBlock {
+    uid: ID!
+    content: String!
+}
+
+# We don't need name, is_enabled, locations, ga_creative, catalog_price_rule_ids, cart_price_rules_ids, locations on storefront
+
+type DynamicBlocksFilterInput {
+    type: DynamicBlockTypeEnum!
+    locations: [DynamicBlockLocationEnum!] # Blocks for all locations will be displayed if not supplied
+    rotation_mode: DynamicBlockRotationModeEnum!
```

Comment:
Why is `rotation_mode` required?

Reply:
rotation_mode comes from data-rotate attribute on the dynamic block [node](https://github.com/magento/architecture/pull/451/files#diff-be24d6f88382fddca4640aefa47a7e508bb0c9913fe7aac6d31814ccd9e34c92R12). Depending on this setting you want to request one random banner, multiple to display all at once, etc.

Reply:
Ah, right - ok, thanks!